x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/schemas/response/schemas.ts

```diff
@@ -25,6 +25,12 @@ export const file_name = t.string;
  */
 export const filters = t.array(t.unknown); // Filters are not easily type-able yet
 
+/**
+ * TODO: Right now the actions is an "unknown", when it could more than likely
+ * become the actual AlertAction as a type.
+ */
+export const actions = t.array(t.unknown); // Actions are not easily type-able yet
```

Comment:
I would not do this without at least trying to type it a bit. I know I didn't do it with filters ... but ...

Really looking at the code you can have these types from here:
https://github.com/elastic/kibana/blob/master/x-pack/plugins/alerting/README.md#post-apialert-create-alert

<img width="882" alt="Screen Shot 2020-03-10 at 10 40 54 AM" src="https://user-images.githubusercontent.com/1151048/76337014-6a98e480-62bc-11ea-887b-429f29ada572.png">

Looking at the code you can have it like so and you can see me playing with it:

<img width="496" alt="Screen Shot 2020-03-10 at 10 45 51 AM" src="https://user-images.githubusercontent.com/1151048/76337066-7ab0c400-62bc-11ea-88db-8f8d113b5596.png">


So what I would here and on the inputs is type it something like:

```ts
/**
 * Params is an "object", since it is a type of AlertActionParams which is action templates.
 * @see x-pack/plugins/alerting/common/alert.ts
 */
export const action = t.exact(
  t.type({ group: t.string, id: t.string, action_type_id: t.string, params: t.object })
);

export const actions = t.array(action);
```

I would add test scripts to this PR in the scripts folder to test these so we can ensure we are not allowing more camel case to "leak through". But it does look like we are possibly leaking more camel case when we are trying to avoid it.

Reply:
Per conversations with alerting team and looking at their API you might not need the action_type_id and if not, then that would be great and you can leave off `action_type_id` but that would up to you as you write a test script if that is possible to exercise this ability.

Reply:
Thank you @FrankHassanabad for great suggestions I hope I've applied them correctly:
https://github.com/elastic/kibana/pull/59641/files#diff-2ce70ffd01576843ee6da6aaa4e2106aR32

https://github.com/elastic/kibana/pull/59641/files#diff-d517059c65ca9b9640932d8fe332ed30R118

https://github.com/elastic/kibana/pull/59641/files#diff-ab0f21e1f43f9aec2258e02c11c839aeR1421